UMS/doc/man/man3/listMachine.3
'\" t
.\"     Title: listMachine
.\"    Author:  SysFera SAS <info@sysfera.com>
.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
.\"      Date: February 2011
.\"    Manual: UMS C++ API Reference
.\"    Source: VISHNU 1.2
.\"  Language: English
.\"
.TH "LISTMACHINE" "3" "February 2011" "VISHNU 1.2" "UMS C++ API Reference"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
listMachine \- lists the machines that are accessible through VISHNU
.SH "SYNOPSIS"
.HP \w'int\ vishnu::listMachine('u
.BI "int vishnu::listMachine(const\ string&\ " "sessionKey" ", ListMachines&\ " "listMachine" ", const\ ListMachineOptions&\ " "options\ =\ ListMachineOptions()" ");"
.SH "DESCRIPTION"
.PP
This command is used to display the machines that you can use for VISHNU services\&. The machines you can access through VISHNU are those that are configured in VISHNU by the VISHNU administrator, and that have been added to your personal VISHNU configuration using the vishnu_add_local_account command\&. The results contain, for each machine, a machine identifier that you can use as a parameter for other VISHNU commands\&.
.SH "ARGUMENTS"
.PP
\fIsessionKey\fR
.RS 4
Input argument\&. The sessionKey is the encrypted identifier of the session generated by VISHNU\&.
.RE
.PP
\fIlistMachine\fR
.RS 4
Output argument\&. ListLocalAccount is the list of the local configs \&.
.RE
.PP
\fIoptions\fR
.RS 4
Input argument\&. Allows a user to list all VISHNU machines or information about a specific machine and an admin to list machines used by a specific user\&.
.RE
.SH "EXCEPTIONS"
.PP
The following exceptions may be thrown:
.PP
"Vishnu not available (Service bus failure)" [1]
.RS 4
.RE
.PP
"Vishnu not available (Database error)" [2]
.RS 4
.RE
.PP
"Vishnu not available (Database connection)" [3]
.RS 4
.RE
.PP
"Vishnu not available (System)" [4]
.RS 4
.RE
.PP
"Internal Error: Undefined exception" [9]
.RS 4
.RE
.PP
"The userId is unknown" [21]
.RS 4
.RE
.PP
"The session key is unrecognized" [28]
.RS 4
.RE
.PP
"The sessionKey is expired\&. The session is closed\&." [29]
.RS 4
.RE
.SH "AUTHOR"
.PP
\fB SysFera SAS\fR <\&info@sysfera.com\&>
.RS 4
Developed the first version of these man pages as part of the VISHNU project.
.RE
.SH "COPYRIGHT"
.br
Copyright \(co 2011 SysFera SAS
.br
.PP
These manual pages are provided under the following conditions:
.PP
Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
.RE
.PP
This software is governed by the CECILL licence under French law and abiding by the rules of distribution of free software. You can use, modify and/ or redistribute the software under the terms of the CeCILL license as circulated by CEA, CNRS and INRIA at the following URL "http://www.cecill.info".
.PP
As a counterpart to the access to the source code and rights to copy, modify and redistribute granted by the license, users are provided only with a limited warranty and the software's author, the holder of the economic rights, and the successive licensors have only limited liability.
.PP
In this respect, the user's attention is drawn to the risks associated with loading, using, modifying and/or developing or reproducing the software by the user in light of its specific status of free software, that may mean that it is complicated to manipulate, and that also therefore means that it is reserved for developers and experienced professionals having in-depth computer knowledge. Users are therefore encouraged to load and test the software's suitability as regards their requirements in conditions enabling the security of their systems and/or data to be ensured and, more generally, to use and operate it in the same conditions as regards security.
.sp


UMS/src/cli/listMachine.cc


#include "listMachine.hh"
#include "utils.hh"

namespace po = boost::program_options;

using namespace std;

int main (int ac, char* av[]){


		/******* Parsed value containers ****************/

		string dietConfig;

		std::string userId;

		std::string machineId;

		std::string sessionKey;

		/********** EMF data ************/

		UMS_Data::ListMachines lsMachine;

		UMS_Data::ListMachineOptions listOptions;

/**************** Describe options *************/


		Options opt(av[0] );

		opt.add("version,v",
				"print version message",
				GENERIC );

        opt.add("dietConfig,c",
						            "The diet config file",
												ENV,
												dietConfig);

        opt.add("userId,u",
						             "An admin option for listing machines in which a specific user has a local configuration",
												CONFIG,
												userId);

				opt.add("listAllmachine,a",
		                    "An option for listing all VISHNU machines",
										    CONFIG);

				opt.add("machineId,i",
									    	"An option for listing information about a	specific machine",
												CONFIG,
												machineId);

				opt.add("sessionKey",
												"The session key",
												ENV,
												sessionKey);

	try {

/**************  Parse to retrieve option values  ********************/

		opt.parse_cli(ac,av);

		bool isEmpty=opt.empty();//if no value was given in the command line

		opt.parse_cfile();

		opt.parse_env(env_name_mapper());

		opt.notify();





/********  Process **************************/


		if (opt.count("listAllmachine")){

			cout <<"We nedd to list all VISHNU machines " << endl;

			listOptions.setListAllmachine(true);
		}


		if (opt.count("userId")){

			cout <<"The user identifier is " << userId << endl;

			listOptions.setUserId(userId);

		}

		if(opt.count("machineId")){

			cout <<"the machineId is : " << machineId << endl;

			listOptions.setMachineId(machineId);
		}

		if(opt.count("sessionKey")){

			cout <<"the session key is : " << sessionKey << endl;
		}

		if (opt.count("dietConfig")){

			cout <<"The diet config file " << dietConfig << endl;
		}
		else{

			cerr << "Set the VISHNU_CONFIG_FILE in your environment variable" <<endl;

			return 1;
		}


		if ( opt.count("help")){

			cout << "Usage: " << av[0] <<" [options]  "<<endl;

			cout << opt << endl;

			return 0;
		}




/************** Call UMS connect service *******************************/


               // initializing DIET

							  if (diet_initialize(dietConfig.c_str(), ac, av)) {
                    cerr << "DIET initialization failed !" << endl;
               return 1;
              }



							int res = listMachine(sessionKey,lsMachine,listOptions);

      							// Display the list
     if(isEmpty) {
         cout << lsMachine << endl;
     }
     else {
         for(int i = 0; i < lsMachine.getMachines().size(); i++) {
            cout << lsMachine.getMachines().get(i) << endl;
          }
     }

	}// End of try bloc

	catch(std::exception& e){
		cout << e.what() <<endl;
		return 1;
	}

	return 0;

}// end of main


